### PR TITLE
Redirect to survey on uninstall

### PR DESCRIPTION
**Pull request by alohaas**
**Reviewer:**
@jdorweiler 

## Description:
Adding a redirect on uninstall

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

**Comment by alohaas** [comment]
@jdorweiler I need to create a new survey monkey URL (the part of the url before the ATB delta parameter), but the rest is ready for review.
Thanks!

File: manifest.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
       }
     ],
     "permissions": [
+        "alarms"
```
Comment (jdorweiler) [inline manifest.json]: missing a comma

File: js/background.js [hunk 2]
```diff
@@ -28,31 +67,24 @@ function Background() {
 
   localStorage['os'] = os;
 
+  chrome.runtime.setUninstallURL('https://www.surveymonkey.com/r/DOC_0', function(){
```
Comment (jdorweiler) [inline js/background.js]: the callback function is optional so you can just get rid of it.

File: js/background.js [hunk 3]
```diff
@@ -28,31 +67,24 @@ function Background() {
...
-            timeSinceEpoch = new Date().getTime() - epoch,
-            majorVersion = Math.ceil(timeSinceEpoch / oneWeek),
-            minorVersion = Math.ceil(timeSinceEpoch % oneWeek / oneDay);
+        var majorVersion = $this.majorVersion();
+            minorVersion = $this.minorVersion();
 
         localStorage['atb'] = 'v' + majorVersion + '-' + minorVersion;
+        localStorage['majorVersion'] = majorVersion;
+        localStorage['minorVersion'] = minorVersion;
+        localStorage['uninstallURL'] = 'https://www.surveymonkey.com/r/DOC_2'
```
Comment (jdorweiler) [inline js/background.js]: What's this for?

File: js/background.js [hunk 4]
```diff
@@ -102,6 +134,20 @@ function Background() {
 
 var background = new Background();
 
+chrome.alarms.create({periodInMinutes: 1});
```
Comment (jdorweiler) [inline js/background.js]: The first param to `create` is an optional name. 
https://developer.chrome.com/apps/alarms#method-create

I would give this alarm a name and then in the listener check to see that we're only responding to that alarm.

File: manifest.json (hunk 1 above)
Comment (alohaas) [inline manifest.json]: oops! thanks

File: js/background.js (hunk 2 above)
Comment (alohaas) [inline js/background.js]: alrighty

File: js/background.js (hunk 3 above)
Comment (alohaas) [inline js/background.js]: ah I was playing around with this and forgot to remove it.  i'll take it out.

File: js/background.js (hunk 4 above)
Comment (alohaas) [inline js/background.js]: sounds good 🆗

**Comment by alohaas** [comment]
@jdorweiler I updated the survey URLs to reflect the correct collectors on survey monkey. Unless there is some more cleanup you'd like me to do, this is ready to be deployed.

File: js/background.js [hunk 5]
```diff
@@ -102,6 +131,17 @@ function Background() {
 
 var background = new Background();
 
+chrome.alarms.create('updateUninstallURL', {periodInMinutes: 1});
+
+chrome.alarms.onAlarm.addListener(function(updateUninstallURL){
```
Comment (jdorweiler) [inline js/background.js]: The param to the function is the actual alarm event. So you want to do something like 
```
function(alarmEvent){
    if(alarmEvent.name === "updateUninstallURL";
   .....
```

**Comment by jdorweiler** [comment]
Looks good. Thanks.

**Pull request merged**